### ISSUE TITLE
zetteldeft-org-graph-note: Wrong type argument: char-or-string-p, nil

### ISSUE THREAD

**Issue by stampf** [EFLS/zetteldeft#58]
Hi,
I tried to create a graph:
in the scratch buffer, or even i a markdown file (my preferred zettelkasten format), I did M-x zetteldeft-org-graph-note, then selected my top §INDEX.md file and got the error in title.
In the buffer, the following was inserted just before the error:
```
#+BEGIN_SRC dot :file ./graph.pdf :cmdline -Kfdp -Tpdf
  
 graph {

  // base note and links 
  "
```

**Comment by EFLS** [comment]
Just to be clear: you need to use this in an Org-mode buffer, because you need to execute the source block. So you can do this from the scratch buffer, but make sure it is in the correct mode.

Anyway, this is unrelated to the error you are getting. I think it is caused by the fact that your `INDEX.md` file doesn't contain a zetteldeft ID, which the function tries to insert.

Moreover, it needs an ID as it uses these as identifiers so that `graphviz` knows which nodes needs to be connected.

I should probably include a check or something, but the graph functions are really experimental...

**Comment by EFLS** [comment]
This should be fixed in 0beb169de3db8c5a4fdc42ce85e8ac6a609f9fc6.

Let me know if there is still an issue.
